src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx

```diff
@@ -197,18 +197,11 @@ class ComparisonFilterUi extends React.Component<ComparisonFilterProps, Comparis
    * Stores the appropriate operator as member.
    */
   onOperatorChange = (newOperator: ComparisonOperator) => {
-
+    let filter: GsComparisonFilter = _cloneDeep(this.state.filter);
+    filter[0] = newOperator;
```

Comment:
don't we need to clone? here and below?

Reply:
The `cloneDeep` is just need when manipulating the "immutable" state.